src/libs/antares/study/parts/common/cluster_list.cpp

```diff
@@ -109,66 +103,30 @@ Data::ClusterList<ClusterT>::~ClusterList()
 template<class ClusterT>
 void ClusterList<ClusterT>::clear()
 {
-    byIndex.clear();
-    if (not cluster.empty())
-        cluster.clear();
-}
-
-template<class ClusterT>
-const ClusterT* ClusterList<ClusterT>::find(const ClusterT* p) const
-{
-    auto end = cluster.end();
-    for (auto i = cluster.begin(); i != end; ++i)
-    {
-        if (p == i->second.get())
-            return i->second.get();
-    }
-    return nullptr;
-}
-
-template<class ClusterT>
-ClusterT* ClusterList<ClusterT>::find(const ClusterT* p)
-{
-    auto end = cluster.end();
-    for (auto i = cluster.begin(); i != end; ++i)
-    {
-        if (p == i->second.get())
-            return i->second.get();
-    }
-    return nullptr;
+    clusters.clear();
 }
 
 template<class ClusterT>
 void ClusterList<ClusterT>::resizeAllTimeseriesNumbers(uint n)
 {
     assert(n < 200000); // arbitrary number
-    if (not cluster.empty())
-    {
-        if (0 == n)
-        {
-            each([&](Cluster& cluster) { cluster.series.timeseriesNumbers.clear(); });
-        }
-        else
-        {
-            each([&](Cluster& cluster) { cluster.series.timeseriesNumbers.resize(1, n); });
-        }
-    }
+    if (n == 0)
```

Comment:
We don't need to separate cases **n ==0** and **n != 0** : 
Indeed, if **n == 0** and we call `resize(1, n)`, then method **clear(...)** is called anyway (please see **resize(...)**'s definition in **Matrix**).
So I guess : 
```C++
template<class ClusterT>
void ClusterList<ClusterT>::resizeAllTimeseriesNumbers(uint n)
{
    assert(n < 200000); // arbitrary number
    each([&](Cluster& cluster) { cluster.series.timeseriesNumbers.resize(1, n); });
}
```
is enough.

Reply:
We can even wonder if **each(...)** is useful, given that **std** contains now all kinds of facilities.
By the way, looking at *each(...)**'s content, it's a simple call to an **std** function.
Don't know if we can remove it easily though.
And it's the same for functions : **begin(...)**, **end(...)**, **empty()**, **size()**, ...